### PR TITLE
feat: withdraw

### PR DESCRIPTION
**Pull request by cazala**
Start a withdraw (sending a meta tx) and send MANA on matic (also via meta tx).

I also added the deposits and withdrawals to the state, and the sagas keep the status of them (pending/checkpoint/complete).

### CODE REVIEW

File: src/components/HomePage/AccountCard/AccountCardHeader/AccountCardHeader.tsx [hunk 1]
```diff
@@ -45,18 +45,16 @@ const AccountCardHeader = ({
...
-            {amount}
+            {parseInt(amount.toFixed(0), 10).toLocaleString()}
           </div>
         </div>
         <div className="actions">
           <Button>
             <Icon name="plus" />
           </Button>
-          <Button onClick={() => onConvertToMaticMana()}>
+          <Button onClick={() => onConvert(network)}>
```
Comment (nachomazzara) [inline src/components/HomePage/AccountCard/AccountCardHeader/AccountCardHeader.tsx:57]: Should we need to have 

```
handleConvert() {
      onReceive(network)
}

<Button onClick={handleConvert}>
```

I'm not sure as it is a one-liner if you prefer using a lambda

File: src/components/HomePage/AccountCard/AccountCard.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ const AccountCard = ({
   const handleOnSeeAll = () => onSeeAll(transactions)
   return (
     <Card className="AccountCard">
-      <AccountCardHeader type={type} title={title} amount={amount} />
+      <AccountCardHeader network={type} title={title} amount={amount} />
```
Comment (nachomazzara) [inline src/components/HomePage/AccountCard/AccountCard.tsx:29]: Should `type` be named as `network`?

File: src/components/HomePage/HomePage.tsx [hunk 3]
```diff
@@ -1,30 +1,24 @@
...
+const HomePage: React.FC<Props> = () => {
   return (
     <>
       <Navbar />
       <Page className="HomePage">
         <AccountHeader />
         <AccountCardContainer>
           <AccountCard
-            type={NetworkType.ETHEREUM}
+            type={Network.ETHEREUM}
```
Comment (nachomazzara) [inline src/components/HomePage/HomePage.tsx:20]: Should `type` be named as `network`?

File: src/components/HomePage/HomePage.tsx [hunk 4]
```diff
@@ -1,30 +1,24 @@
...
-  Props,
-  TransactionStatus,
-  TransactionType,
-} from './HomePage.types'
+import { Props, TransactionStatus, TransactionType } from './HomePage.types'
 import './HomePage.css'
 
-const HomePage: React.FC<Props> = (props) => {
-  const { mana, manaL2 } = props
+const HomePage: React.FC<Props> = () => {
```
Comment (nachomazzara) [inline src/components/HomePage/HomePage.tsx:12]: I didn't check out the branch but, Is `Props` necessary here?

File: src/components/HomePage/HomePage.types.tsx [hunk 5]
```diff
@@ -1,14 +1,6 @@
...
-export type Props = {
-  mana: number
-  manaL2: number
-}
-
-export enum NetworkType {
-  MATIC = 'matic',
-  ETHEREUM = 'ethereum',
-}
+export type Props = {}
```
Comment (nachomazzara) [inline src/components/HomePage/HomePage.types.tsx:3]: Based on the above comment. Is also needed here?

File: src/components/Modals/ConvertManaModal/ConvertManaModal.css [hunk 6]
```diff
@@ -0,0 +1,92 @@
+.ui.modal.ConvertManaModal {
+  padding: 26px;
+  width: 520px;
+}
+.ConvertManaModal .title {
```
Comment (nachomazzara) [inline src/components/Modals/ConvertManaModal/ConvertManaModal.css:5]: blank missed

File: src/components/Modals/ConvertManaModal/ConvertManaModal.css [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  font-style: normal;
+  font-weight: normal;
+  font-size: 15px;
+  line-height: 24px;
+  margin-top: -15px;
+  margin-bottom: 10px;
+
+  /* identical to box height, or 160% */
+  text-align: center;
+
```
Comment (nachomazzara) [inline src/components/Modals/ConvertManaModal/ConvertManaModal.css:35]: Should we need to remove this blank?

File: src/components/Modals/ConvertManaModal/ConvertManaModal.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+  const handleSetAmount = (e: React.FormEvent<HTMLInputElement>) => {
+    const intValue = parseInt(e.currentTarget.value, 10)
+    if (e.currentTarget.value.length === 0) {
+      setAmount(0)
+    } else if (!isNaN(intValue)) {
+      setAmount(intValue)
+    }
+  }
+  const getMaximumValue = () => toBN(2).pow(toBN(255))
```
Comment (nachomazzara) [inline src/components/Modals/ConvertManaModal/ConvertManaModal.tsx:34]: Maybe we can move this to a constant

File: src/components/Modals/ReceiveManaModal/ReceiveManaModal.tsx [hunk 9]
```diff
@@ -28,7 +28,7 @@ const ReceiveManaModal = ({ name, onClose, metadata }: ModalProps) => {
       <Modal.Content>
         <div className="message"> {t('receive_mana_modal.message')} </div>
         <Field
-          value={`${walletAddress.slice(0, 6)}...${walletAddress.slice(-4)}`}
+          value={`${address.slice(0, 6)}...${address.slice(-4)}`}
```
Comment (nachomazzara) [inline src/components/Modals/ReceiveManaModal/ReceiveManaModal.tsx:31]: have we something similar at `decentraland-dapps`?

File: src/modules/mana/sagas.ts [hunk 10]
```diff
@@ -52,16 +109,16 @@ function* handleDepositManaRequest(action: DepositManaRequestAction) {
   const { amount } = action.payload
 
   try {
-    const eth: Eth | null = yield call(createEth)
-    if (!eth) {
-      throw new Error('Could not get Eth')
+    const provider = yield call(getConnectedProvider)
+    if (!provider) {
+      throw new Error(`Could not get connected provider`)
```
Comment (nachomazzara) [inline src/modules/mana/sagas.ts:114]: We can change `  to '  if you want

File: src/modules/mana/sagas.ts [hunk 11]
```diff
@@ -73,19 +130,49 @@ function* handleDepositManaRequest(action: DepositManaRequestAction) {
...
+}
+
 function* handleGetApprovedManaRequest(_action: GetApprovedManaRequestAction) {
   try {
-    const eth: Eth | null = yield call(createEth)
-    if (!eth) {
-      throw new Error('Could not get Eth')
+    const provider = yield call(getConnectedProvider)
+    if (!provider) {
+      throw new Error(`Could not get connected provider`)
```
Comment (nachomazzara) [inline src/modules/mana/sagas.ts:173]: Same here

File: src/modules/mana/sagas.ts [hunk 12]
```diff
@@ -109,10 +196,11 @@ function* handleGetApprovedManaRequest(_action: GetApprovedManaRequestAction) {
 function* handleApproveManaRequest(action: ApproveManaRequestAction) {
   const { allowance } = action.payload
   try {
-    const eth: Eth | null = yield call(createEth)
-    if (!eth) {
-      throw new Error('Could not get Eth')
+    const provider = yield call(getConnectedProvider)
+    if (!provider) {
+      throw new Error(`Could not get connected provider`)
```
Comment (nachomazzara) [inline src/modules/mana/sagas.ts:201]: and here

File: src/modules/mana/utils.ts [hunk 13]
```diff
@@ -1,6 +1,94 @@
...
+        type: 'function',
+      },
+    ] as any,
+    provider
+  )
+}
+
+export async function isWithdrawalSynced(txHash: string) {
+  const provider = new ethers.providers.JsonRpcProvider(
+    'https://rpc-mumbai.matic.today'
```
Comment (nachomazzara) [inline src/modules/mana/utils.ts:44]: Should we need to move this to an env or get it from `decentraland-dapps` or `decentraland-transactions`?

File: src/modules/mana/utils.ts [hunk 14]
```diff
@@ -1,6 +1,94 @@
...
+  )
+
+  const isSynced = checkpoints.length > 0
+  return isSynced
+}
+
+export async function isDepositSynced(txHash: string) {
+  // get root counter
+  const provider = new ethers.providers.JsonRpcProvider(
+    'https://goerli.infura.io/v3/faae5316a4604118badb59ee6a01de28'
```
Comment (nachomazzara) [inline src/modules/mana/utils.ts:63]: Same as above about the RPC URL

**Review by nachomazzara — CHANGES_REQUESTED** [review]
Excellent! this is historic :rocket:

Just minor comments

**Review by malaniz — APPROVED** [review]
LGTM!

File: src/modules/mana/utils.ts (hunk 13 above)
Comment (cazala) [reply src/modules/mana/utils.ts:44]: ahh sorry! yes this shouldn't be hardcoded haha

File: src/modules/mana/utils.ts (hunk 14 above)
Comment (cazala) [reply src/modules/mana/utils.ts:63]: yep totally, this shouldn't be here 😅

File: src/components/Modals/ReceiveManaModal/ReceiveManaModal.tsx (hunk 9 above)
Comment (cazala) [reply src/components/Modals/ReceiveManaModal/ReceiveManaModal.tsx:31]: there was a helper in the marketplace for this but not in dapps

Comment (nachomazzara) [reply src/components/Modals/ReceiveManaModal/ReceiveManaModal.tsx:31]: should we need to create an issue to move it to dapps (both places)?

**Review by nachomazzara — APPROVED** [review]

File: src/components/Modals/ReceiveManaModal/ReceiveManaModal.tsx (hunk 9 above)
Comment (cazala) [reply src/components/Modals/ReceiveManaModal/ReceiveManaModal.tsx:31]: sure, let's add an issue to decentraland-dapps and i ll add it for the next release

**Pull request merged**